src/extensionWeb.ts

```diff
@@ -28,63 +17,21 @@ export async function activate(context: vscode.ExtensionContext) {
     )
 
     try {
-        // Setup the logger
-        const toolkitOutputChannel = vscode.window.createOutputChannel('AWS Toolkit', { log: true })
-        await activateLogger(context, toolkitOutputChannel)
-
-        setupGlobals()
-
-        await initializeComputeRegion()
-        initialize(context)
-        initializeManifestPaths(context)
-
-        await activateTelemetry(context, globals.awsContext, Settings.instance)
-
-        await initializeCredentials(context, globals.awsContext, globals.loginManager)
-        await initializeAwsCredentialsStatusBarItem(globals.awsContext, context)
-
-        registerCommands(context)
+        // IMPORTANT: Any new activation code should be done in the function below unless
+        // it is browser specific activation code.
+        await activateShared(context, () => {
+            return {
+                guessDefaultRegion: () => defaultRegion,
+            } as RegionProvider
+        })
     } catch (error) {
         const stacktrace = (error as Error).stack?.split('\n')
         // truncate if the stacktrace is unusually long
         if (stacktrace !== undefined && stacktrace.length > 40) {
             stacktrace.length = 40
         }
-        getLogger().error('Failed to activate extension in Browser', error)
-        throw error
+        getLogger().error(`Failed to activate extension`, error)
     }
 }
 
-/** Set up values for the `globals` object */
-function setupGlobals() {
```

Comment:
Looks like this change eliminated some unessential variation between the "regular" vs browser case. And that's exactly the point of doing this now: unwinding those non-essential varations _later_ would be difficult.